braindecode/models/eegconformer.py

```diff
@@ -0,0 +1,272 @@
+    -----------------------------
+    - fc_dim: int
+        The dimension of the fully connected layer.
+    - n_classes: int
+        Number of classes.
+
+
+    References
+    ----------
+    .. [EEG Conformer] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       IEEE Transactions on Neural Systems and Rehabilitation Engineering,
+       31, pp.710-719. https://ieeexplore.ieee.org/document/9991178
+    .. [EEG Conformer Code] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       https://github.com/eeyhsong/EEG-Conformer.
+
+
+    """
+
+    def __init__(
+            self,
+            n_kernel=40,
+            n_kernel_temp_conv=25,
+            n_kernel_spatial_conv=22,
+            n_kernel_avg_pool=75,
+            stride_avg_pool=15,
+            drop_prob=0.5,
+            att_depth=6,
+            att_heads=10,
+            att_drop_prob=0.5,
+            fc_dim=2440,
+            n_classes=4,
+    ):
+        super().__init__(
+            PatchEmbedding(
+                n_kernel,
+                n_kernel_temp_conv,
+                n_kernel_spatial_conv,
+                n_kernel_avg_pool,
+                stride_avg_pool,
+                drop_prob,
+            ),
+            TransformerEncoder(att_depth, n_kernel, att_heads, att_drop_prob),
+            ClassificationHead(n_kernel, fc_dim, n_classes),
+        )
+
+
+class PatchEmbedding(nn.Module):
+    """Patch Embedding.
+
+    The authors used a convolution moduleto capture local features,
+    instead of postion embedding.
+
+    """
+
+    def __init__(
+            self,
+            n_kernel,
+            n_kernel_temp_conv,
+            n_kernel_spatial_conv,
+            n_kernel_avg_pool,
+            stride_avg_pool,
+            drop_prob,
+    ):
+        super().__init__()
+
+        self.shallownet = nn.Sequential(
+            nn.Conv2d(1, n_kernel, (1, n_kernel_temp_conv), (1, 1)),
+            nn.Conv2d(n_kernel, n_kernel, (n_kernel_spatial_conv, 1), (1, 1)),
+            nn.BatchNorm2d(n_kernel),
+            nn.ELU(),
+            nn.AvgPool2d(
+                (1, n_kernel_avg_pool), (1, stride_avg_pool)
+            ),
+            # pooling acts as slicing to obtain 'patch' along the
+            # time dimension as in ViT
+            nn.Dropout(drop_prob),
+        )
+
+        self.projection = nn.Sequential(
+            nn.Conv2d(
+                n_kernel, n_kernel, (1, 1), stride=(1, 1)
+            ),  # transpose, conv could enhance fiting ability slightly
+            Rearrange("b d_model 1 seq -> b seq d_model"),
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = x.unsqueeze(dim=1)  # add one extra dimension
+        x = self.shallownet(x)
+        x = self.projection(x)
+        return x
+
+
+class MultiHeadAttention(nn.Module):
```

Comment:
If this module and the rest below are used only internally in this file they should be private classes with `_` as first letter in the name, e.g., `_MultiHeadAttention`